task3

xtaide opened the pull request:
Ещё немного поправил listLength, чтоб правильно показывал всплывающее окошко, не знаю, правильно ли брать длину именно так как я её взял, но отображает нормально.

Commit:
task3

Ещё немного поправил listLength, чтоб правильно показывал всплывающее окошко, не знаю, правильно ли брать длину именно так как я её взял, но отображает нормально.

NikitaPripotnev commented on #3/src/components/customField/CustomSelect.js:
```diff
@@ -0,0 +1,152 @@
+							height: parameters.height + 'rem',
+						}}
+					>
+						{groups === false ? (
```
Я так понимаю тут должно быть group, а не groups, будь внимательнее, уже второй раз на неделе такой косяк, первый раз был с переменной arr. Проверяй работу компонента при разных параметрах!
Тем не менее, отталкиваться от group === false нельзя в данном случае (да и наверное ни в каком), потому что у нас есть написанные ранее инпуты, в которых параметра group нет как такого, да и указывать его в кажом случае когда ты хочешь использовать обычный селект отстой. А если его не указываешь, то приходит undefined, а undefined === false будет false и пойдет он у тебя на отрисовку Groups и в итоге вообще ничего не выведет, так как там еще одна проверка.
Делаешь проверку на group === true, а на проверку groups.length можно забить 

Также ты убрал items(заменил по сути на groups) из параметров и снова старые инпуты не будут работать, потому что они передают параметр items)) И тут к тебе вопросики, так как ради того, чтобы работало со старыми пропсами ты тернарный оператор вроде как делал на вилку Groups и Items, а по факту все это работать не будет

NikitaPripotnev commented on #3/src/components/customField/CustomSelect.js:
```diff
@@ -0,0 +1,152 @@
+				.getComputedStyle(document.documentElement, null)
+				.getPropertyValue('font-size')
+		)
+		const listHeight = Math.min(groups[0].items.length * 2.8 + 0.8, 16.25)
```
Тут тебе пришлось менять коэфициент, потому что раньше можно было посчитать заранее размер попапа со списком, отталкиваясь от line-height и количества айтемов. Собственно и сейчас это можно сделать, не увеличивая коэфициент на случайную величину, а прибавляя размер тайтлов групп. Но тут ты вообще просто берешь первую группу и ее размер, по сути просто не понимая, что это за формула и что она делает.
Исправлять или доделывать это ненужно, я сделал похитрее (наверное) через useLayout, не считая размер вручную. Тебе предстоит все это оформить в какой-то классный кастомный хук на след неделе)

NikitaPripotnev commented on #3/src/components/customField/CustomField.scss:
```diff
@@ -0,0 +1,216 @@
+	flex-direction: column;
+	justify-content: left;
+	text-align: left;
+	width: 250px;
```
Вот это сомнительная вещь, прям под конкретную ситуацию, у нас эти инпуты могут растягиваться по разному, поэтому фикса в px, еще и не в относительных величинах типо em - не приветствуется

NikitaPripotnev reviewed (commented):
Из-за того что выкладывал как проект react, файлы поменяли свое местоположение и на реквесте я не вижу что именно было изменено, но ничего страшного, при работе в проекте такого не будет)
Из-за причины описанной выше не вчитывался в файлы, так, что бросилось в глаза - расписал. Уделяй больше времени на проверку компонента.

Commit:
#3 rework customField

Теперь работает с любыми items)

NikitaPripotnev commented on #3/src/components/customField/CustomSelect.js:
```diff
@@ -30,6 +29,20 @@ const CustomSelect = ({
 	const parent = useRef(null)
 
+	// Сразу узнаю всё про группы, если что-то есть, возвращаю массив с количеством итемов в каждом объекте, которые передаются
+	const group = items[0].values
```
Вот это все уже ближе к истине, но: 
1) Не переназначай локальную переменную в методах array.prototype типо map, если хочешь вернуть массив длин, то возвращай из map просто map(item=>item.values.length+1) - так как у тебя всегда есть хэдер, то прибавляешь +1
2) Не оч красивая конструкция, что у тебя возвращается либо булево значение, либо число
3) Я стараюсь не делать таких локальных переменных в компоненте, которые будут каждый рендер вычисляться, хотя по сути они зависят только от пропса items

NikitaPripotnev reviewed (commented):
Я бы по сути создал стэйт локальный с айтемами типо itemsLocal, и через useEffect(()=>, [items]) его наполнял, и возможно это бы был объект с айтемами, длиной, и флагом group

Commit:
#3 useLayoutEffect on popup

Использовал useLayoutEffect

Commit:
#3 upgraded select

Убрал из отдельных файлов группы и итемы. С хуком ничего в голову не приходит. Понимание как писать хук есть, но как в данном случае применить не понимаю :cry:
